statsmodels/stats/multicollinearity.py

```diff
@@ -0,0 +1,460 @@
+    others.
+
+    see class MultiCollinearSequential where all variables are analyzed in given sequence.
+
+    Parameters
+    ----------
+    data : array_like
+        data with observations in rows and variables in columns. This is ignored if
+        moment_matrix is not None.
+    moment_matrix : None or ndarray
+        Optional moment or correlation matrix to be used for the analysis. If it is provided
+        then data is ignored.
+    standardize : bool, default True
+        If true and data are provided, then the data will be standardized to mean zero and
+        standard deviation equal to one, which is equivalent to using the correlation matrix.
+        TODO: This might be replaced by separate demean option. Variance inflation factor
+        can be calculated for data that is scaled but not demeaned.
+        See Notes
+
+    Notes
+    -----
+    The default treatment assumes that there is no constant in the data array or in the
+    moment matrix. However, VIF is defined under the assumption of a constant in the
+    corrsponding regression. Use `standardize=False` if demeaning is not desired.
+
+    THe only case I know so far is if we have a categorical factor in the design matrix but
+    no constant, that is the constant is implicit in the full dummy set. Using VIF with demeaned
+    data will show that the dummy set is perfectly collinear (up to floating point precision).
+
+
+    """
+
+    def __init__(self, data, moment_matrix=None, standardize=True):
+
+        if hasattr(data, 'columns'):
+            self.columns = data.columns
+        else:
+            self.columns = None
+
+        if data is not None:
+            x = np.asarray(data)
+
+        # TODO: use pandas corrcoef below to have nan handling ?
+
+        if moment_matrix is not None:
+            xm = np.asarray(moment_matrix)
+            if standardize:
+                # check if we have correlation matrix,
+                # we cannot demean but we can scale
+                # (We could demean if const is included in uncentred moment matrix)
+                if (np.diag(xm) != 1).any():
+                    xstd = np.sqrt(np.diag(xm))
+                    xm = xm / np.outer(xstd, xstd)
+        else:
+            if standardize:
+                xm = np.corrcoef(x, rowvar=0)
+                if np.any(np.ptp(x, axis=0) == 0):
+                    # TODO: change this? detect from xcorr, drop constant ?
+                    raise ValueError('If standardize is true, then data should ' +
+                                     'not include a constant')
+            else:
+                xm = np.dot(x.T, x) / float(x.shape[0])
+
+        self.k_vars = xm.shape[1]
+        self.mom = xm
+
+    @cache_readonly
+    def vif(self):
+        """Variance inflation factor based on moment or correlation matrix.
+        """
+        # np.diag returns read only array, need copy
+        vif_ = np.diag(np.linalg.inv(self.mom)).copy()
```

Comment:
make inv(mom) a cached attribute
then we can get diag of it directly (e.g. without thresholding)